hub/api/tests/test_api.py

```diff
@@ -53,6 +53,6 @@ def test_iterate_dataset(ds):
     for idx, sub_ds in enumerate(ds):
         img = sub_ds["image"].numpy()
         label = sub_ds["label"].numpy()
-        np.testing.assert_array_equal(img, np.ones((1, 28, 28)))
-        assert label.shape == (1, 1)
-        assert label[0] == labels[idx]
+        np.testing.assert_array_equal(img, np.ones((28, 28)))
+        assert label.shape == (1,)
```

Comment:
`__iter__` uses integer indexing, so the generated samples should not have the batch axis

Reply:
do we want to eventually support batched iteration here? i kinda feel like no because that's where pytorch/tensoflow come in.

Reply:
Yeah, `__iter__` should not be responsible for this. We can accumulate samples externally via PT/TF or any other wrapper